pkg/build/api/types.go

```diff
@@ -111,6 +111,10 @@ type BuildSource struct {
 	// This allows to have buildable sources in directory other than root of
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
+
+	// SourceSecretName is the name of a Secret that would be used for setting
+	// up the authentication for cloning private repository.
+	SourceSecretName string `json:"sourceSecretName,omitempty"`
```

Comment:
@bparees @smarterclayton one question that pop-ed up in my head today, why not using `ObjectReference` for that? I've taken here the approach @mfojtik did when doing secrets for docker push, but wouldn't it be desirable to use `ObjectReference` to mark this field as using external object?